src/middlewared/middlewared/api/v25_04_0/catalog.py

```diff
@@ -0,0 +1,104 @@
+from datetime import datetime
+
+from pydantic import ConfigDict, Field, RootModel
+
+from middlewared.api.base import BaseModel, ForUpdateMetaclass, NonEmptyString, single_argument_args, LongString
+
+
+__all__ = [
+    'CatalogEntry', 'CatalogUpdateArgs', 'CatalogUpdateResult', 'CatalogTrainsArgs', 'CatalogTrainsResult',
+    'CatalogSyncArgs', 'CatalogSyncResult', 'CatalogAppInfo', 'CatalogAppsArgs', 'CatalogAppsResult',
+    'CatalogAppDetailsArgs', 'CatalogAppDetailsResult',
+]
+
+
+class CatalogEntry(BaseModel):
+    id: NonEmptyString
+    label: NonEmptyString = Field(pattern=r'^\w+[\w.-]*$')
+    preferred_trains: list[NonEmptyString]
+
+
+@single_argument_args('catalog_update')
+class CatalogUpdateArgs(BaseModel, metaclass=ForUpdateMetaclass):
+    preferred_trains: list[NonEmptyString]
+
+
+class CatalogUpdateResult(BaseModel):
+    result: CatalogEntry
+
+
+class CatalogTrainsArgs(BaseModel):
+    pass
+
+
+class CatalogTrainsResult(BaseModel):
+    result: list[NonEmptyString]
+
+
+class CatalogSyncArgs(BaseModel):
+    pass
+
+
+class CatalogSyncResult(BaseModel):
+    result: None
+
+
+class Maintainer(BaseModel):
+    name: str
+    email: str
+    url: str | None
+
+
+class CatalogAppInfo(BaseModel):
+    app_readme: LongString | None = Field(description='HTML content of the app README.')
```

Comment:
Okay, so you can get around all these `Field(` definitions by adding a doctoring underneath each variable. The docstrings will be rendered in the api docs automatically. i.e. something like
```
app_readme: LongString | None = None
"""HTML content of the app README."""
categories: list[str]
"""List of categories for the app."""